src/dialect/snowflake.rs

```diff
@@ -0,0 +1,26 @@
+use crate::dialect::Dialect;
+
+#[derive(Debug, Default)]
+pub struct SnowflakeDialect;
```

Comment:
I'd like the new dialect (including a separate test file), the dialect-specific parsing infrastructure, and the table factor parsing fix to be landed separately.

Reply:
You mean by different PR's or commits ?

Reply:
I meant PRs.